File: app/src/main/java/com/wavesplatform/wallet/ui/dex/details/chart/ChartModel.java
package com.wavesplatform.wallet.ui.dex.details.chart;

import android.databinding.BaseObservable;

import com.github.mikephil.charting.data.CombinedData;
import com.wavesplatform.wallet.payload.Candle;
import com.wavesplatform.wallet.payload.WatchMarket;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class ChartModel extends BaseObservable {


    private List<Candle> candleList;
    private CombinedData data = new CombinedData();
    private Date lastLoadDate;
    private WatchMarket pairModel;

    public ChartModel() {
        candleList = new ArrayList<>();
    }

    public List<Candle> getCandleList() {
        return candleList;
    }

    public void setCandleList(List<Candle> candleList) {
        this.candleList = candleList;
    }

    public CombinedData getData() {
        return data;
    }

    public void setData(CombinedData data) {
        this.data = data;
    }

    public Date getLastLoadDate() {
        return lastLoadDate;
    }

    public void setLastLoadDate(Date lastLoadDate) {
        this.lastLoadDate = lastLoadDate;
    }

    public WatchMarket getPairModel() {
        return pairModel;
    }

    public void setPairModel(WatchMarket pairModel) {
        this.pairModel = pairModel;
    }
}

File: app/src/main/java/com/wavesplatform/wallet/v2/data/model/local/ChartModel.kt
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.wallet.v2.data.model.local

import com.github.mikephil.charting.data.CombinedData
import com.wavesplatform.sdk.model.response.data.CandlesResponse
import com.wavesplatform.wallet.v2.data.model.remote.response.WatchMarketResponse
import java.util.*

class ChartModel  {

    var candleList: List<CandlesResponse.Data.CandleResponse> = ArrayList()
    var data = CombinedData()
    var lastLoadDate: Date = Date()
    var pairModel: WatchMarketResponse? = null
}
